pkg/controllers/sync/version/manager.go

```diff
@@ -70,38 +63,52 @@ type VersionManager struct {
 
 	versions map[string]runtimeclient.Object
 
-	client generic.Client
+	client fedcorev1a1client.CoreV1alpha1Interface
 
 	logger klog.Logger
 }
 
-func NewVersionManager(
+func NewNamespacedVersionManager(
 	logger klog.Logger,
-	client generic.Client,
-	namespaced bool,
-	federatedKind, targetKind, namespace string,
+	client fedcorev1a1client.CoreV1alpha1Interface,
+	namespace string,
 ) *VersionManager {
+	adapter := NewVersionAdapter(true)
 	v := &VersionManager{
-		logger:        logger.WithValues("origin", "version-manager"),
-		targetKind:    targetKind,
-		federatedKind: federatedKind,
-		namespace:     namespace,
-		adapter:       NewVersionAdapter(namespaced),
-		versions:      make(map[string]runtimeclient.Object),
-		client:        client,
+		logger:    logger.WithValues("origin", "version-manager", "type-name", adapter.TypeName()),
+		namespace: namespace,
+		adapter:   adapter,
+		versions:  make(map[string]runtimeclient.Object),
+		client:    client,
+	}
+
+	return v
+}
+
+func NewClusterVersionManager(
+	logger klog.Logger,
+	client fedcorev1a1client.CoreV1alpha1Interface,
+) *VersionManager {
+	adapter := NewVersionAdapter(false)
+	v := &VersionManager{
+		logger:    logger.WithValues("origin", "version-manager", "type-name", adapter.TypeName()),
+		namespace: "",
+		adapter:   adapter,
+		versions:  make(map[string]runtimeclient.Object),
+		client:    client,
 	}
 
 	return v
 }
 
 // Sync retrieves propagated versions from the api and loads it into
 // memory.
-func (m *VersionManager) Sync(stopChan <-chan struct{}) {
-	versionList, ok := m.list(stopChan)
+func (m *VersionManager) Sync(ctx context.Context) {
+	versionList, ok := m.list(ctx)
 	if !ok {
 		return
 	}
-	ok = m.load(versionList, stopChan)
+	ok = m.load(ctx, versionList)
 	if !ok {
 		return
 	}
```

Comment:
In the comment doc for `VersionManager.Get` below, let's also explain that an empty map is returned when the given `VersionedResource` does not match the last version recorded by the `VersionManager`.